connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MySQLIntegrationSuite.scala

```diff
@@ -264,4 +342,10 @@ class MySQLOverMariaConnectorIntegrationSuite extends MySQLIntegrationSuite {
       s"jdbc:mysql://$ip:$port/mysql?user=root&password=rootpass&allowPublicKeyRetrieval=true" +
         s"&useSSL=false"
   }
+
+  // MariaDB Connector/J uses collation 'utf8mb4_unicode_ci' as collation for connection.
+  // The MySQL server 9.1.0 uses collation 'utf8mb4_0900_ai_ci' for database by default.
+  // This method uses string literal so the result of cast and literal have the same collation.
+  override def testCastStringTarget(stringLiteral: String, stringCol: String): String =
```

Comment:
I'm not sure if this function is good enough to test cast string.

Reply:
The root cause is the MariaDB JDBC driver is not able to use the same default collation for connection. If we can accept changing the collation of connection to make it consistent with MySQL server, the test logic can be the same for both MySQL JDBC and MariaDB JDBC.

I didn't want to change the default settings for all tests, but that's probably the only way to fix the inconsistency and use the same code in test. What do you think about this?

Reply:
Or in other words, it is due to compatibility issue between MariaDB JDBC and MySQL Server 9.